server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java

```diff
@@ -876,6 +877,16 @@ private void deduplicate(QueryPlan queryPlan, int fetchSize)
                   .getContext()
                   .getColumnName()
               : columnForReader;
+      if (queryPlan instanceof AggregationPlan && ((AggregationPlan) queryPlan).getLevel() >= 0) {
+        String aggregatePath =
+            originalPath.isMeasurementAliasExists()
+                ? FilePathUtils.generatePartialPathByLevel(
+                    originalPath.getFullPathWithAlias(), ((AggregationPlan) queryPlan).getLevel())
+                : FilePathUtils.generatePartialPathByLevel(
+                    originalPath.toString(), ((AggregationPlan) queryPlan).getLevel());
```

Comment:
In line 850, check the level and don't change the value of `columnForReader`. In this way, you can use the `columnForReader` directly here.

Reply:
The modified value of `columnForReader` is used between line 858-871. Considering the complexity of `deduplicate()`, I tried not to make much changes here to keep correctness. 
Perhaps refactor `deduplicate()` and moving it into QueryPlan would be a choice. What do you think?

Reply:
Okay...but maybe we have to rewrite this method later...it's so long